### ISSUE TITLE
Package request: treedome

### ISSUE THREAD

**Issue by tengkuizdihar** [NixOS/nixpkgs#253033]
**Project description**

A local-first, encrypted, note taking application with tree-like structures, all written and saved in your computer.

**Metadata**

* homepage URL: https://gitlab.com/treedome/treedome
* source URL: https://gitlab.com/treedome/treedome
* license: GPL3+
* platforms: Linux, Darwin, Microsoft Windows

**Comment by tengkuizdihar** [comment]
I'll probably the one that does this, just making an issue to be proper, that's all.

**Comment by jfvillablanca** [comment]
I tried packaging using the upstream flake.
I get a runtime error though

```
DRI driver not from this Mesa build ('23.1.3' vs '23.1.7')
failed to bind extensions
Failed to create GBM device for render device: /dev/dri/renderD128: No such file or directory
Failed to get GBM device
```

`nix run gitlab:treedome/treedome/0.3.2` works fine though so I figured I missed something in any of the phases that I needed to adopt since I am pulling from GitLab

if anyone wants to take a stab, here is the WIP

```nix
{ lib
, cargo-tauri
, cmake
, dbus
, fetchFromGitLab
, fetchYarnDeps
, freetype
, gsettings-desktop-schemas
, gtk3
, libsoup
, mkYarnPackage
, openssl
, pkg-config
, rustPlatform
, webkitgtk
, wrapGAppsHook
, sqlite
}:

let
  pname = "treedome";
  version = "0.3.2";

  srcRoot = fetchFromGitLab {
    owner = "treedome";
    repo = "treedome";
    rev = version;
    hash = "sha256-n/y2UYbJBj1ed2fpPGE6BqUsGY8qUaBsVUMPoovqYjg=";
  };

  frontend-build = mkYarnPackage {
    inherit version; src = srcRoot;
    pname = "treedome-ui";

    offlineCache = fetchYarnDeps {
      yarnLock = srcRoot + "/yarn.lock";
      hash = "sha256-rV5jKKnbMutaG5o8gRKgs/uoKwbIkxAPIcx6VWG7mm4=";
    };

    packageJSON = srcRoot + "/package.json";

    configurePhase = ''
      ln -s $node_modules node_modules
    '';

    buildPhase = ''
      export HOME=$(mktemp -d)
      yarn --offline run build

      mkdir -p $out/dist
      cp -r dist/** $out/dist
    '';

    distPhase = "true";
    dontInstall = true;
  };
in

rustPlatform.buildRustPackage {
  inherit version pname;

  src = srcRoot + "/src-tauri";

  # essential in NVIDIA + compositor https://github.com/NixOS/nixpkgs/issues/212064#issuecomment-1400202079
  WEBKIT_DISABLE_COMPOSITING_MODE = 1;

  cargoLock = {
    lockFile = srcRoot + "/src-tauri/Cargo.lock";
    outputHashes = {
      "fix-path-env-0.0.0" = "sha256-ewE3CwqLC8dvi94UrQsWbp0mjmrzEJIGPDYtdmQ/sGs=";
    };
  };

  # copy the frontend static resources to final build directory
  # Also modify tauri.conf.json so that it expects the resources at the new location
  postPatch = ''
    mkdir -p dist
    cp -R ${frontend-build}/dist/** dist

    ls -al
    substituteInPlace ./tauri.conf.json --replace '"distDir": "../dist",' '"distDir": "dist",'
    substituteInPlace ./tauri.conf.json --replace '"beforeBuildCommand": "yarn run build",' '"beforeBuildCommand": "",'
  '';

  nativeBuildInputs = [ cmake pkg-config cargo-tauri wrapGAppsHook ];
  buildInputs = [ dbus openssl freetype libsoup gtk3 webkitgtk gsettings-desktop-schemas sqlite ];

  buildPhase = ''
    runHook preBuild

    cargo tauri build

    runHook postBuild
  '';

  installPhase = ''
    mkdir -p $out/bin/
    mkdir -p $out/share/

    cp target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/bin/treedome $out/bin/treedome
    cp -R target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/share/** $out/share/
  '';
}
```

**Comment by tengkuizdihar** [comment]
already in nixpkgs
